frappe/custom/doctype/customize_form/customize_form.py

```diff
@@ -570,3 +587,53 @@ def reset_customization(doctype):
 	('Table', 'Table MultiSelect'))
 
 ALLOWED_OPTIONS_CHANGE = ('Read Only', 'HTML', 'Select', 'Data')
+
+
+def genrate_qr_code_file(doc, method):
+	"""
+	Generate and attach QR Code for the document for public access
+	"""
+	# if QR Code field not present, do nothing
+	if not hasattr(doc, 'qr_code'):
+		return
+
+	# Don't create QR Code if it already exists
+	qr_code = doc.get("qr_code")
+	if qr_code and frappe.db.exists({"doctype": "File", "file_url": qr_code}):
+		return
+
+	# Creating public url to print format
+	default_print_format = frappe.db.get_value('Property Setter', dict(property='default_print_format', doc_type=doc.doctype), "value")
+
+	# System Language
+	language = frappe.get_system_settings('language')
+
+	# creating qr code for the url
+	url = f"{ frappe.utils.get_url() }/{ doc.doctype }/{ doc.name }?format={ default_print_format or 'Standard' }&_lang={ language }&key={ doc.get_signature() }"
```

Comment:
is there some specification using which the url was designed this way?
I do not think the user will the user get redirected to this document if he scans this qr code?
Then how will the qrcode be used?
Thanks

Reply:
@hasnain2808 The url pattern has been followed from here:
https://github.com/frappe/frappe/blob/develop/frappe/templates/emails/print_link.html

The QR Code has the url to access the document default print format without login in. Please view the gif for the working

![2021-07-27 at 23 20 05 - QR Code](https://user-images.githubusercontent.com/7881486/127210696-724b8f22-f197-4082-a4f6-f1966aaa5d00.gif)